### ISSUE TITLE
X-Link Container

### ISSUE THREAD

**Issue by arianmares** [jonas-koeritz/hfa-dissector#1]
Do you have any information about the data structure of this?

**Comment by jonas-koeritz** [comment]
I don't have information on this by now, I will rewrite this in the near future as some assumptions have been proven wrong in specific cases. This is useful for general debugging work, but not usable as a full protocol spec. Especially the authentication (hash algorithm) is not known by now.

**Comment by arianmares** [comment]
Your information regarding HFA was the only I've found. I am using it to decode caller/called IDs in a recording software which I develop. Until the last IP phones generation, the information in screen messages was enough, but the last sets with graphic display doesn't use the set display commands but sends these info packed in a X-Link container.  I've managed to find the caller ID in such a container but some more infos would be useful. Thank you.

**Comment by jonas-koeritz** [comment]
The X-Link Containers might contain CSTA (ASN.1) encoded data.

arianmares notifications@github.com schrieb am Di., 21. Juni 2016, 07:33:

**Comment by arianmares** [comment]
I've studied some captured data. All I can say is that the data is definitely not valid BER, CER, or DER, nor is it one of the XML rules (XER, EXER, CXER). It could be PER, but  needs a schema. No idea if this is a path to follow (looking for a schema). Any thoughts?

**Comment by jonas-koeritz** [comment]
I will try to setup a phone on debug log level and make assumptions based
on timing. The basic packet structure is a bit off in the current code too,
I noticed this when reviewing log files of a phone. This must be addressed
in the future too. My code is messy and needs a rewrite too ;)

arianmares notifications@github.com schrieb am Do., 14. Juli 2016, 18:13:

**Comment by arianmares** [comment]
I can send you some pcap files, in some x-link containers some caller ID
data can be identified, but nothing else.
Also, I am not a C++ wizzard but I can do decent coding, if you need
some help, just let me know.

La 07/14/2016 08:08 PM, Jonas Köritz a scris:

**Comment by jonas-koeritz** [comment]
The current code is written in LUA only and I would like to keep it this
way until the protocol is fully reverse engineered. Would you attach the
pcap files and some information on what happened during the capture?

arianmares notifications@github.com schrieb am Do., 14. Juli 2016, 19:44:

**Comment by arianmares** [comment]
Yes, will try to do it  this evening or tomorrow morning.

La 07/14/2016 09:05 PM, Jonas Köritz a scris:

**Comment by jonas-koeritz** [comment]
I got a first idea, the contents could be HDLC with SS7 signalling inside. I will try to verify this tomorrow.

**Comment by jonas-koeritz** [comment]
Another possibility is H.225.0 User to User information elements. This appears to be much more probable!

**Comment by arianmares** [comment]
Could you, please, give me an email address so I can send the pcap file,
the size exceeds the github mailbox.

La 07/15/2016 02:02 AM, Jonas Köritz a scris:

**Comment by arianmares** [comment]
Yes, as I said, looks like ASN.1 PER encoding, but still need a schema...

La 07/15/2016 02:02 AM, Jonas Köritz a scris:

**Comment by jonas-koeritz** [comment]
jonas.koeritz@gmail.com

arianmares notifications@github.com schrieb am Fr., 15. Juli 2016, 07:22:
